device/globals/interactive.gd

```diff
@@ -1,165 +1,36 @@
 extends Node
-export(Script) var animations
 
-var terrain
-var walk_path
-var walk_context
-var moved
-var last_scale = Vector2(1, 1)
-var last_deg = null
-var last_dir = 0
-var animation
-var state = ""
-var walk_destination
-var path_ofs
-var pose_scale = 1
-var task
-var sprites = []
+export var global_id = ""
+export(String, FILE, ".esc") var events_path = ""
+export var active = true setget set_active,get_active
 
-export var speed = 300
-export var scale_on_map = false
-export var light_on_map = false setget set_light_on_map
+var event_table = {}
 
-# This appears to be slightly faster in _process than checking
-# `self is Node2D` or something similar on every loop
-onready var self_has_z_index = self is Node2D
+func register_to_vm():
```

Comment:
This function feels a bit redundant... I mean, calling `vm.register_object(global_id, self)` is pretty obvious if it's spelled out in `_ready()`.

The decision on this is @StraToN 's, but I'm not a huge fan of these types of indirections. The bigger value might be in adding debug prints temporarily, but that doesn't necessarily require a wrapper function. Is there another solid reason?

Reply:
I agree with @mjtorn . If `register_to_vm()` is to be called only in `_ready()` function and only contains a one liner, then directly put this one liner in `_ready()` function